packages/cmf/src/expressions/immutable.js

```diff
@@ -0,0 +1,12 @@
+import _get from 'lodash/get';
+import Immutable from 'immutable';
+import curry from 'lodash/curry';
+
+function getInState(statePath, { context }, immutablePath, defaultValue) {
+	return _get(context.store.getState(), statePath, new Immutable.Map()).getIn(
+		immutablePath.split('.'),
+		defaultValue,
+	);
+}
+
+export default curry(getInState);
```

Comment:
This super selectors seems odd and a little bit complex.
I have imagined selectors  to  be more precise (collectionSelectors, componentSelectors etc).
Not a big fan of this.

Reply:
The first version was written this way and we add exactly the same code with just 'collections' and 'components' difference.

Reply:
Don't you think immutable.js is and odd name for a file ? 
It can be misleading and confusing with the library immutable.